src/main/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticator.java

```diff
@@ -109,8 +117,16 @@ public HTTPJwtAuthenticator(final Settings settings, final Path configPath) {
         jwtHeaderName = settings.get("jwt_header", HttpHeaders.AUTHORIZATION);
         isDefaultAuthHeader = HttpHeaders.AUTHORIZATION.equalsIgnoreCase(jwtHeaderName);
         rolesKey = settings.get("roles_key");
+        jsonRolesPath = settings.get("roles_path");
         subjectKey = settings.get("subject_key");
         jwtParser = _jwtParser;
+
+        // throw exception when settings provided both roles_key and roles_path
+        if (rolesKey != null && jsonRolesPath != null) {
```

Comment:
we also need to make sure that only one of them is set at a time, either rolesKey or jsonRolesPath